### DISCUSSION TITLE
What are the mathematical models implemented in cobrapy?

### DISCUSSION THREAD

**Discussion by galenseilis** [opencobra/cobrapy#1365] [Q&A]
Skimming the [readthedocs](https://cobrapy.readthedocs.io/en/latest/index.html) I get the gist that this package is about modelling the chemical reaction rates based on constraints and chemical reaction equations. 

What isn't clear to me is the family(ies) of mathematical models actually being used. The chemical reaction equations tell me about stoichiometry, but not so much what these rates are. Is the model the [master equation](https://en.wikipedia.org/wiki/Master_equation), or something else?

Thanks,

G

**Comment by galenseilis** [comment]
I also did not see any equations in [the paper](https://bmcsystbiol.biomedcentral.com/articles/10.1186/1752-0509-7-74).

**Comment by cdiener** [comment]
No, no dynamic models. Mostly [constraint-based optimization](https://en.m.wikipedia.org/wiki/Constrained_optimization) in particular flux balance analysis (see https://doi.org/10.1038%2Fnbt.1614).
